src/Tribe/RSVP.php

```diff
@@ -744,10 +760,20 @@ public function generate_tickets() {
 			tribe_exit();
 		}
 
-		// Iterate over each product
-		foreach ( (array) $_POST['product_id'] as $product_id ) {
-			if ( ! $ticket_qty = $this->parse_ticket_quantity( $product_id ) ) {
-				// if there were no RSVP tickets for the product added to the cart, continue
+		$product_ids = [];
+
+		if ( isset( $_POST['tribe_ticket_id'] ) ) {
+			$product_ids = (array) $_POST['tribe_ticket_id'];
+		} elseif ( isset( $_POST['product_id'] ) ) {
+			$product_ids = (array) $_POST['product_id'];
```

Comment:
* :warning: Detected usage of a non-sanitized input variable: $_POST['product_id']
* :no_entry_sign: Processing form data without nonce verification.

<sub>hash: eb3d9485f9281cc02225ab76bb7de36d</sub>